anvill/python/anvill/arch.py

```diff
@@ -35,6 +35,9 @@ def return_stack_pointer_proto(self, num_bytes_popped):
     def pointer_size(self):
         raise NotImplementedError()
 
+    def stack_offset(self):
```

Comment:
Can you import `abc` and then make `Arch` derive from `ABC`, then change all these things throwing `NotImplementedError` to use the `@abstractmethod` decorator, then make the function bodies `...`, e.g.

```python
@abstractmethod
def foo():
    ...
```

The `...` is actually itself there.

Reply:
Also, add return type hints where clear/obvious.